CompactedSyntax/CompactedPDFSyntaxTest.pdf.json

```diff
@@ -0,0 +1,1658 @@
+[
```

Comment:
This JSON appears not to document the trailer - is that correct?

Reply:
And "Null" should really be "null"

Reply:
Resolved

Reply:
> This JSON appears not to document the trailer - is that correct?

This becomes a matter of defining the scope of the lexical analyser (e.g. does it include decryption and object stream parsing?). After the indirect objects have been parsed and identified by object number, the trailer has no meaning anymore.

Reply:
The trailer in the test PDF does have a very specific parsing test (comment after name without whitespace) that is not duplicated elsewhere which is why I asked.

And no matter which way you look at or name things, something somewhere has to lex and parse the startxref, trailer(s), xref(s), incremental updates, etc. before you get to knowing anything about potential decrypting (which itself requires lexing/parsing of certain key data structures) which is before you eventually get to lexing/parsing the majority of main body objects in a PDF. The PDF spec is also not prescriptive about things such as lazy resolution, when and in what order you lex, parse or validate certain syntactic constructs, etc. - arguably this is just one reason "shadow attacks" are successful.

I've no problem in accepting your JSON contribution (which I think is really useful as a human-understandable version) so long as such caveats and assumptions are clearly documented as not all lexer/parsers use the same assumptions, hence why I was asking about the originating source. 

Maybe the simpler solution is just to put this documentation into the README.md?

Reply:
Thank you. I agree. I will update the README.

Reply:
> And "Null" should really be "null"

To be consistent with the casing of null, I have pushed a new version that formats JSON properties and PDF object types lower-case.